engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/text/util/TypeToDisplayTextNode.java

```diff
@@ -7,33 +7,46 @@
 import org.enso.interpreter.runtime.type.TypesGen;
 
 import com.oracle.truffle.api.CompilerDirectives;
-import com.oracle.truffle.api.dsl.GenerateUncached;
-import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.dsl.NeverDefault;
 import com.oracle.truffle.api.interop.InteropLibrary;
 import com.oracle.truffle.api.interop.UnsupportedMessageException;
-import com.oracle.truffle.api.library.CachedLibrary;
 import com.oracle.truffle.api.nodes.Node;
 
-@GenerateUncached
-public abstract class TypeToDisplayTextNode extends Node {
-  public abstract String execute(Object o);
+public final class TypeToDisplayTextNode extends Node {
+  @Child
+  private TypeOfNode typeOfNode;
+
+  private TypeToDisplayTextNode(TypeOfNode typeOfNode) {
+    this.typeOfNode = typeOfNode;
+  }
+
+  public String execute(Object o) {
+    return switch (o) {
+      case Type t -> t.getName();
+      default -> switch (typeOfNode.execute(o)) {
+        case Type t -> t.getName();
```

Comment:
What is the reason for adding this fallback?

Reply:
`TypeToDisplayTextNode` currently operates on `Type` (and gives its simple type name) as well as on values - then it finds the type and gives its simple type name.

Reply:
I see. Maybe we can have a comment at the top of this class stating what the capabilities of this node are? (That it can give the simple name of a type value, but also resolve a type of a value and give _its_ simple type name)

Reply:
I don't want to add a comment and explain a behavior I don't understand. I had no idea why the node handles types in a special way. It is not easy to understand the intentions. Usually I am just driven by the desire to pass existing unit tests...

However with faf7005 I believe this "unexplainability" is gone. It seems that most of the special casing was caused by `AnyToDisplayTextNode` (waiting for unit tests to finish to verify). If so, it is more logical to add `@Specialization` into `AnyToDisplayTextNode` and let the `TypeToDisplayTextNode` handle just "simple type name" of an object without any exceptions.